calcom/cal.com
# Fix: non existent page causes a 404 with no feedback

## What does this PR do?

The pull request adds up a custom 404 error page to enhance user feedback for nonexistent pages and adjusts routing to display this page for invalid URLs. It features a homepage link and a search bar to improve navigation.

- Fixes #15956 

## Mandatory Tasks (DO NOT REMOVE)




## How should this be tested?

- Verify Display: Navigate to invalid URLs to ensure the custom 404 page appears correctly.

- Test Navigation: Check that the homepage link and search bar function as expected.

- Review Routing: Confirm routing changes trigger the custom 404 page for undefined routes.


- Yes, ensure environment variables for API keys, routing configurations, and build settings are correctly set.

- The minimal test data includes valid and invalid URLs to check the 404-page display and navigation elements.
